src/nvim/testdir/test_join.vim

```diff
@@ -52,9 +51,8 @@ func Test_join_marks()
 
   /^This line/;'}-join
   call assert_equal([0, 4, 11, 0], getpos("'["))
-  call assert_equal([0, 4, 67, 0], getpos("']"))
+  call assert_equal([0, 4, 66, 0], getpos("']"))
   enew!
-  set nojoinspaces
```

Comment:
keep the testdir/ tests same as upstream where possible. These changes don't look necessary (unless they are actually closer to upstream now)